### PR TITLE
add REST endpoint swagger documentation

### PR DESCRIPTION
**Pull request by wjrjerome**
image
image
image
image

image

### CODE REVIEW

**Review by liam-lai — DISMISSED** [review]

File: swagger.yaml [hunk 1]
```diff
@@ -1,20 +1,238 @@
...
+        400:
+          description: 'Bad Request'
+        409:
+          description: 'Conflict'
         500:
           description: 'Server Error'
-
+  /information/masterNodes:
+    get:
+      summary: Get master nodes information such as committ size and their status
```
Comment (liam-lai) [inline swagger.yaml:22]: committed

File: swagger.yaml [hunk 2]
```diff
@@ -1,20 +1,238 @@
...
+                    properties:
+                      committeeSize:
+                        type: integer
+                      activeNodeSize:
+                        type: integer
+                      standByNodeSize:
+                        type: integer
+                    required:
+                      - committeeSize
+                      - activeNodeSize
```
Comment (liam-lai) [inline swagger.yaml:45]: I think we don't need `Size` as subfix

File: swagger.yaml [hunk 3]
```diff
@@ -1,20 +1,238 @@
...
+definitions:
+  MasterNode:
+    type: object
+    properties:
+        address:
+          description: The master node xdc address in subnet
+          type: string
+        role:
+          type: string
+          enum: [STANDBY, COMMITTEE_MEMBER]
```
Comment (liam-lai) [inline swagger.yaml:226]: [STANDBY, MINER, PENALTY]

File: swagger.yaml [hunk 4]
```diff
@@ -1,20 +1,238 @@
...
+                      balance:
+                        description: The super/admin account remaining balance in XDC
+                        type: string
+                      walletAddress:
+                        description: The wallet address of the account
+                        type: string
+                    required:
+                      - walletAddress
+                      - balance
+                  relayer:
```
Comment (liam-lai) [inline swagger.yaml:82]: I think we can just have backlog and contractAddress at first layer.

**Comment by liam-lai** [comment]
/information/relayer
 /information/network
 /information/masterNodes
 Need `status` field

File: swagger.yaml [hunk 5]
```diff
@@ -1,20 +1,238 @@
...
+                    required:
+                      - block
+                  parentChain:
+                    type: object
+                    properties:
+                      targetChain:
+                        description: A string value which is used to identify the target parent chain. It can be a URL, IP address or a name.
+                        type: string
+                    required:
+                      - targetChain
```
Comment (liam-lai) [inline swagger.yaml:136]: I think we have flat format:
{
   averageBlockTime
   txThroughput
   targetChain
}

**Comment by liam-lai** [comment]
Do you think we can have all 3 information api merge into 1 api? Seems like each API has not many information inside. And pull out masternodes list as it's `currentHeight` is so dynamic, can consider push api.

File: swagger.yaml (hunk 5 above)
Comment (wjrjerome) [reply swagger.yaml:136]: nah. there are mixed information here. some belong to subnet and some pointing to the actual parent chain network. In the case of adding more fields in the future, i recommend keep it with nested layer.

**Comment by wjrjerome** [comment]
> 

I would say keep it seperate for now. if we are to merge them together, it won't be much work. but it's better keep them seperate for now so that we can work in parallel.

**Pull request merged**